R/content-tools.R

```diff
@@ -44,35 +42,62 @@ extract_tool_requests <- function(contents) {
   contents[is_tool_request]
 }
 
+tool_result_factory <- function(tool, arguments, id) {
+  function(result = NULL, error = NULL) {
```

Comment:
Good practice to `force(tool)` etc here: https://adv-r.hadley.nz/function-factories.html#forcing-evaluation